File: backend/src/main/java/com/socialmeet/backend/entity/PaymentMethod.java
package com.socialmeet.backend.entity;

/**
 * 支付方式枚举
 */
public enum PaymentMethod {
    ALIPAY("alipay", "支付宝"),
    WECHAT("wechat", "微信支付"),
    UNIONPAY("unionpay", "银联支付");

    private final String code;
    private final String displayName;

    PaymentMethod(String code, String displayName) {
        this.code = code;
        this.displayName = displayName;
    }

    public String getCode() {
        return code;
    }

    public String getDisplayName() {
        return displayName;
    }

    public static PaymentMethod fromCode(String code) {
        for (PaymentMethod method : values()) {
            if (method.code.equalsIgnoreCase(code)) {
                return method;
            }
        }
        throw new IllegalArgumentException("Unknown payment method code: " + code);
    }
}

File: app/src/main/java/com/example/myapplication/model/PaymentMethod.kt
package com.example.myapplication.model

/**
 * 支付方式枚举
 */
enum class PaymentMethod(val code: String, val displayName: String, val iconRes: Int) {
    ALIPAY("alipay", "支付宝", com.example.myapplication.R.drawable.ic_alipay),
    WECHAT("wechat", "微信支付", com.example.myapplication.R.drawable.ic_wechat);
    
    companion object {
        /**
         * 根据代码获取支付方式
         */
        fun fromCode(code: String): PaymentMethod? {
            return values().find { it.code == code }
        }
        
        /**
         * 根据代码获取支付方式，如果找不到则返回默认值
         */
        fun fromCodeOrDefault(code: String, default: PaymentMethod = ALIPAY): PaymentMethod {
            return fromCode(code) ?: default
        }
    }
}

